docs: release notes for v0.10.0 (#246)
ricochet-rs/docs: src/content/docs/releases/v0.10.0.mdx

---
title: v0.10.0
docs: release notes for Ricochet v0.10.0
slug: releases/v0.10.0
tableOfContents: false
template: splash
---

**Released: June 8, 2026**

### Features

> **UI: Smooth Page Transitions**: Pages now transition smoothly with a persistent shell and cross-fade animation for a more polished user experience.

Reviewer: ```suggestion
**UI: Smooth Page Transitions**: Pages now transition more smoothly with a cross-fade animation for a more polished user experience. Note that it respects accessibility preferences as well.
```

**Auth: Global Admin Override**: Global administrators can now override collaborator mutations for better access management.

**Auth: OIDC Group Sync**: Added support for synchronizing groups from Microsoft Graph via OIDC, enabling seamless integration with Microsoft Entra ID.

> **Proxy: Container Backend Improvements**: The Docker backend has been renamed to "container" and now supports execution environments with bind mounts for more flexible deployment options.

Reviewer: ```suggestion
**Container Backend Improvements**: The Docker backend has been renamed to "container" and now supports execution environments with bind mounts for more flexible deployment options.
```

### Fixes

**Auth: OIDC Group Access**: Fixed OIDC group access to work correctly with Microsoft Entra and improved error diagnostics.

**UI: Edge Browser Freeze**: Resolved an issue that caused Microsoft Edge to freeze on the app loading page.

> **Proxy: Deployment Cancellation**: Unblocked deployment cancellation so users can stop deployments as expected.

Reviewer: ```suggestion
**Deployment Cancellation**: Unblocked deployment cancellation so users can stop deployments as expected.
```

> **Proxy: Image Alias Validation**: Fixed image alias passing to execution environment validation.

Reviewer: ```suggestion
**Image Alias Validation**: Fixed image alias passing to execution environment validation.
```

> **API: Connection Failure Handling**: Ricochet no longer attempts to start if the API fails to connect, preventing startup issues.

Reviewer: ```suggestion
**Connection Failure Handling**: Ricochet no longer attempts to start if the API fails to connect, preventing startup issues.
```
